### DISCUSSION TITLE
Authelia, Caddy and Subpath configuration documentation

### DISCUSSION THREAD

**Discussion by tzfx** [authelia/authelia#5660] [Troubleshooting] [answered]
**TLDR**: Caddy-Authelia [subpath configuration](https://www.authelia.com/integration/proxies/caddy/#subpath) requires the `server.path` prop to be set in authelia's config-- is this something that I should put a docs PR up to make clear?

---

More context:

I recently setup Authelia with Caddy for the first time following all the documentation and it took me the better part of the afternoon to realize that the 'subpath' configuration <https://www.authelia.com/integration/proxies/caddy/#subpath> fails to mention that the `server.path` config option (in authelia's configuration) needs to be set to match the auth subpath. Otherwise, the portal's base tag points to the root of the domain and the supporting files (js,css,etc) fail to load.

This seems obvious in retrospect, however even finding that there *was* a `server.path` config in the configuration.yml options that updated the base path had me diving through code.

Subdomains make for happy SNI, but if we have subpath config documentation it is probably worth updating the docs to outline that one additional config that needs to be updated.

**Comment by james-d-elliott** [comment, marked as answer]
Yeah a PR is fine, try to make sure you do it for every proxy with a subpath config example and that it is an identiccal note similar to the other `Important Note` sections we have elsewhere.

**Comment by tzfx** [comment]
Oh my gosh, I thought I was going crazy a bit. So, when I originally started setting this up, I opted to use docker. I didn't realize that `authelia/authelia` (latest) is using **v4.37.5**:

```bash
~ $ docker run --rm authelia/authelia authelia --help
Unable to find image 'authelia/authelia:latest' locally
latest: Pulling from authelia/authelia
# ...snip
6925530fdee2: Pull complete 
Digest: sha256:25fc5423238b6f3a1fc967fda3f6a9212846aeb4a720327ef61c8ccff52dbbe2
Status: Downloaded newer image for authelia/authelia:latest
authelia v4.37.5
# ...snip
```

This version (the `latest` tag, by the look of it) does not have support for `server.address` (or `session.cookies`) at all. This clears up *a lot* of the confusion I was having, as the example configuration.yml options for docker-compose (in master) fail to work out of the box due to this split version situation.

That is- The docker-compose files checked into master at `examples/compose/(lite|local)/docker-compose.yml` both use `authelia/authelia` as the image, when really they should be changed to `authelia/authelia:master`. Otherwise, the example configuration.yml files fail to validate and the container goes into a restart loop.

So uh, what's the deal with that? I guess this should all clear up when the next version of authelia is tagged and released to docker on :latest?

I'll work the doc PR but I my initial intention was to document setting `path` at `server.path` when it's clear (now) that I should be setting (and telling others to set) the `path` part of `server.address:tcp://:9091/<path>` sort of thing.

**Reply by james-d-elliott** [reply]
They will not be changed to `authelia/authelia:master`. It will not be the default image for any example as this will lead to a support nightmare.

The instructions in the readme points you to the docs here which instruct you how to checkout the code if you want to do that which work: https://www.authelia.com/integration/deployment/docker/

The configuration files are in sync with the changes on master, otherwise we'd have to implement all of those changes all at once instead of incrementally.

**Reply by tzfx** [reply]
> The instructions in the readme points you to the docs here which instruct you how to checkout the code if you want to do that which work: ...

🤦 I see what happened-- The docs specifically have us swap to the latest tag (which I clearly skipped), which would effectively "downgrade" the example configurations to be compatible with the tagged release. Thanks for clearing that up!
